tldw_Server_API/tests/Agent_Client_Protocol/test_mcp_adapter.py

```
@@ -0,0 +1,337 @@
+        "mcp_transport": "stdio",
+        "command": "echo",
+        "mcp_orchestration": "llm_driven",
+    })
+
+    with patch(
+        "tldw_Server_API.app.core.Agent_Client_Protocol.adapters.mcp_adapter.create_transport",
+        return_value=mock_transport,
+    ):
+        await adapter.connect(config)
+
+    with pytest.raises(ValueError, match="llm_driven.*llm_caller.*tool_gate"):
```

Ruff (0.15.6)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)